### ISSUE TITLE
Is Operation state with empty actions array valid?

### ISSUE THREAD

**Issue by MarianMacik** [serverlessworkflow/specification#776]
Specification says that the `actions` array of the Operation state [is required](https://github.com/serverlessworkflow/specification/blob/main/specification.md#operation-state). Schema also [lists it](https://github.com/serverlessworkflow/specification/blob/main/schema/workflow.json#L864-L887) as a required property.

However, there seems to be no constraint on the size of the array, so technically an Operation state with an empty `actions` array is valid. This also [validates with the Java SDK](https://github.com/serverlessworkflow/sdk-java/blob/main/api/src/main/resources/schema/states/operationstate.json#L20-L26). However, with the Go SDK [the kubebuilder tag and also the Go validator tag say](https://github.com/serverlessworkflow/sdk-go/blob/main/model/operation_state.go#L28-L29) that the minimum size of the `actions` array is 1, so an empty array is refused.

So is the Operation state with an empty `actions` array valid?

**What would you like to be added**:
Clarification on the size of the `actions` array in the schema and in the specification document.

**Why is this needed**:
To not have discrepancies among different SDKs.

Adding @ricardozanini @spolti to the loop too.

**Comment by tsurdilo** [comment]
> there seems to be no constraint on the size of the array, so technically an Operation state with an empty actions array is valid

yep that is true, the property itself is required but can be empty array. 

> Clarification on the size of the actions array in the schema and in the specification document.

Imho its pretty clear from spec and schema that it can be empty array?

> To not have discrepancies among different SDKs.

Yeah sdk should allow empty array

**Comment by ricardozanini** [comment]
@tsurdilo so we can fix the SDK Go. But in what context we can have an Operation State with an empty actions array?

**Comment by cdavernas** [comment]
> yep that is true, the property itself is required but can be empty array.

But does that make sense? Shouldn't `actions` require at least one action? Otherwise, what's the point of it?

**Comment by ricardozanini** [comment]
We should agree on:

1. Required array fields are ok to be empty
2. When empty, tooling/implementations could warn based on the context
3. Do not change the schemas

If we agree, we do nothing since the schema already reflects that (it's required, but an empty array is valid). It was a decision on SDK Go to fail on an empty array that we should revert and respect the schema instead. cc @serverlessworkflow/sdk-developers

**Comment by MarianMacik** [comment]
This was fixed by serverlessworkflow/sdk-go#183 which corrected the kubebuilder tag to allow an empty array.
cc @ricardozanini
